include/veriblock/pop/blockchain/base_block_tree.hpp

```diff
@@ -849,21 +850,22 @@ struct BaseBlockTree {
   virtual bool finalizeBlockImpl(index_t& index,
                                  // see config.preserveBlocksBehindFinal()
                                  int32_t preserveBlocksBehindFinal,
-                                 ValidationState& state) {
+                                 ValidationState& /*state*/) {
     VBK_TRACE_ZONE_SCOPED;
     VBK_LOG_DEBUG("Finalize %s, preserve %d blocks behind",
                   index.toShortPrettyString(),
                   preserveBlocksBehindFinal);
 
     index_t* finalizedBlock = &index;
 
-    // prereq is not met - finalized block must be on active chain
-    if (!activeChain_.contains(finalizedBlock)) {
-      return state.Invalid("block-not-on-active-chain");
+    if (finalizedBlock == getRoot()) {
+      return true;
     }
```

Comment:
Why this check?

Reply:
It's an early return

Reply:
Yes but it is covered but the code below. Does not look like it makes any speed up.

Reply:
this case will be triggered often, and instead of iterating over `tips` and doing some other looping we could just early return, this is faster by definition (we don't invoke code we don't need)